IMove.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace objected_oriented_programming
+{
+    public interface IMove
```

Comment:
Название интерфейса обычно выбирается как "объект, поддерживающий некоторые возможности" (в переводе на английский звучит лаконичнее).
```suggestion
    public interface IMoveable
```

Reply:
Ну либо `ICoder` - тоже сойдёт. Но `Move` - это действие. Не очень годится в качестве названия интерфейса. Хотя, опять же, всё зависит от контекста задачи.